python/plugins/processing/algs/gdal/viewshed.py

```diff
@@ -0,0 +1,158 @@
+*   (at your option) any later version.                                   *
+*                                                                         *
+***************************************************************************
+"""
+
+__author__ = 'Alexander Bruy'
+__date__ = 'October 2019'
+__copyright__ = '(C) 2019, Alexander Bruy'
+
+import os
+
+from qgis.PyQt.QtGui import QIcon
+
+from qgis.core import (QgsRasterFileWriter,
+                       QgsProcessingException,
+                       QgsProcessingParameterDefinition,
+                       QgsProcessingParameterRasterLayer,
+                       QgsProcessingParameterBand,
+                       QgsProcessingParameterPoint,
+                       QgsProcessingParameterNumber,
+                       QgsProcessingParameterString,
+                       QgsProcessingParameterRasterDestination)
+from processing.algs.gdal.GdalAlgorithm import GdalAlgorithm
+from processing.algs.gdal.GdalUtils import GdalUtils
+
+from processing.tools.system import isWindows
+
+pluginPath = os.path.split(os.path.split(os.path.dirname(__file__))[0])[0]
+
+
+class viewshed(GdalAlgorithm):
+
+    INPUT = 'INPUT'
+    BAND = 'BAND'
+    OBSERVER = 'OBSERVER'
+    OBSERVER_HEIGHT = 'OBSERVER_HEIGHT'
+    TARGET_HEIGHT = 'TARGET_HEIGHT'
+    MAX_DISTANCE = 'MAX_DISTANCE'
+    OPTIONS = 'OPTIONS'
+    EXTRA = 'EXTRA'
+    OUTPUT = 'OUTPUT'
+
+    def __init__(self):
+        super().__init__()
+
+    def initAlgorithm(self, config=None):
+        self.addParameter(QgsProcessingParameterRasterLayer(self.INPUT,
+                                                            self.tr('Input layer')))
+        self.addParameter(QgsProcessingParameterBand(self.BAND,
+                                                     self.tr('Band number'),
+                                                     1,
+                                                     parentLayerParameterName=self.INPUT))
+        self.addParameter(QgsProcessingParameterPoint(self.OBSERVER,
+                                                      self.tr('Observer location')))
+        self.addParameter(QgsProcessingParameterNumber(self.OBSERVER_HEIGHT,
+                                                       self.tr('Observer height'),
+                                                       type=QgsProcessingParameterNumber.Double,
+                                                       minValue=0.0,
+                                                       defaultValue=1.0))
+        self.addParameter(QgsProcessingParameterNumber(self.TARGET_HEIGHT,
+                                                       self.tr('Target height'),
+                                                       type=QgsProcessingParameterNumber.Double,
+                                                       minValue=0.0,
+                                                       defaultValue=1.0))
+        self.addParameter(QgsProcessingParameterNumber(self.MAX_DISTANCE,
```

Comment:
Same with this - what unit is it? Maybe a distance parameter is more suitable here

Reply:
Changed to distance parameter